.github/workflows/test.yml

```diff
@@ -51,14 +51,9 @@ jobs:
         run: |
           set -x
           go version
-          export GOMAXPROCS=2
-          args=("-p" "2" "-race")
-          # Run with less concurrency on Windows to minimize flakiness.
-          if [[ "${{ matrix.platform }}" == windows* ]]; then
-            unset args[2]
-            args[1]="1"
-            export GOMAXPROCS=1
-          fi
+          # Run with less concurrency to reduce CI flakiness.
+          export GOMAXPROCS=1
```

Comment:
I'm not sure about this change. I think it makes sense since when we see a flakey tests we don't have the resources to resolve them straight away, and they can be quite demanding tests to fix. At the same time, i think we've found some issues in CI that we don't see in our local test runs, which are generally race conditions or deadlocks. I'll go with what the consensus is on this one 🙂

Reply:
That's a good point. I wonder if we should have one set of tests specifically run with higher concurrency, or maybe just leave the current setup and expect it to pass in any case.

Reply:
This is the one that makes it consistently (eh, to some degree) pass. Otherwise, it constantly fails.

Reply:
When I'm testing this PR with several commits (in and out), I noticed, the nature of the tests don't matter for the outcome of the test runs.